### PR TITLE
Adds a new workflow to automatically publish releases

### PR DESCRIPTION
**Pull request by not-jenni**
This workflow is triggered after the "Build Native Release Packages" workflow succeeds, and publishes a release that was created as a draft in the "Schedule Snapshot Release" workflow.

Sample workflow run at  https://github.com/not-jenni/iree/runs/3304500983?check_suite_focus=true, which generated https://github.com/not-jenni/iree/releases/tag/snapshot-20210811.29

The action will not run if any of the builds fail (tested in https://github.com/not-jenni/iree/runs/3304557762?check_suite_focus=true), and more thorough validation of the packages will be added soon.

This uses https://github.com/marketplace/actions/publish-release.

Progress on https://github.com/google/iree/issues/6315 and https://github.com/google/iree/issues/6316

### CODE REVIEW

File: .github/workflows/build_package.yml [hunk 1]
```diff
@@ -251,3 +251,16 @@ jobs:
         with:
           release_id: ${{ github.event.inputs.release_id }}
           assets_path: ./bindist/*
+
+  validate_and_publish:
+    name: "Invoke workflow to validate and publish release"
+    needs: build_core
```
Comment (ScottTodd) [inline .github/workflows/build_package.yml:257]: How about we also flip the `experimental: true` bit to false for those builds that are currently passing?

For example, 
* https://github.com/google/iree/runs/3271008495?check_suite_focus=true failed on `macos-latest :: Build main-dist Package`
* https://github.com/google/iree/runs/3299480891?check_suite_focus=true failed on `windows-2019 :: Build main-dist Package`

and both of those normally succeed (so we should mark them as non-experimental and triage/fix issues, or deflake the builds as needed)

File: .github/workflows/schedule_snapshot_release.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ jobs:
           body: |
             Automatic snapshot release of iree.
           draft: true
-          prerelease: true
+          prerelease: false
```
Comment (ScottTodd) [inline .github/workflows/schedule_snapshot_release.yml:49]: @stellaraccident any reservations about dropping the "prerelease" label on releases and making all GitHub Releases snapshots "full"/"official" releases? We can still publish stable releases through a different channel (or multiple channels, e.g. PyPI for wheels, Maven Central for Java/Android, vcpkg for C++) with a directly controlled index page.

**Review by ScottTodd — COMMENTED** [review]
Nice! Thanks for the details in the description and testing the failure mode. Just a few comments, but then I think this looks good.

File: .github/workflows/schedule_snapshot_release.yml (hunk 2 above)
Comment (not-jenni) [reply .github/workflows/schedule_snapshot_release.yml]: Turns out the action still works if snapshots are marked as pre-release, so I reverted this line

File: .github/workflows/build_package.yml [hunk 3]
```diff
@@ -55,7 +55,7 @@ jobs:
           # Windows packages.
           - os: windows-2019
             build_package: main-dist
-            experimental: true
+            experimental: false
           - os: windows-2019
             build_package: py-runtime-pkg
             experimental: true
```
Comment (ScottTodd) [inline .github/workflows/build_package.yml:61]:
```suggestion
            experimental: false
```

https://github.com/google/iree/actions/runs/1099349568 here's a build where what normally works builds successfully

only "experimental" builds are windows py-xla, windows py-tflite, and windows py-tf by that definition

File: .github/workflows/build_package.yml [hunk 4]
```diff
@@ -72,7 +72,7 @@ jobs:
           # Macos packages.
           - os: macos-latest
             build_package: main-dist
-            experimental: true
+            experimental: false
           - os: macos-latest
             build_package: py-runtime-pkg
             experimental: true
```
Comment (ScottTodd) [inline .github/workflows/build_package.yml:78]: false (see my other comment), same for the other builds below

**Review by ScottTodd — APPROVED** [review]
I'm in favor of merging this to get an release attempt tonight, but it seems like the macos main-dist package may be broken (due to mlir-hlo, recent logs: https://github.com/google/iree/runs/3316641122?check_suite_focus=true), so that may need to be fixed before a manual (or automated) release gets published

**Comment by ScottTodd** [comment]
oops, disabled automerge since I had some comments to address and it ignores those :p
lgtm once a few more builds are flipped to non-experimental

**Pull request merged**